### PR TITLE
Refactor Haskell Code

### PR DESCRIPTION
**Pull request by Ian-Stewart-Binks**
This pull request minorly refactors some of the existing Haskell code. Meant as an incremental refactoring step.

Most of the changes that you will notice are indentation changes. Much of the code in the WebParsing module used an indentation style inconsistent with the rest of the codebase.

Some of the naming has been improved. You will see both files and variables renamed.

There have been other minor fixes as well.

### CODE REVIEW

File: hs/WebParsing/TimeTableParser.hs [hunk 1]
```diff
@@ -14,17 +14,17 @@ import Database.JsonParser
...
 -- from the table. Is is later converted into lecture or tutorial records by examinig the
 -- first letter of the section.
 data CourseSlot =
-  CourseSlot {
-              slotSection :: T.Text,
-              slotTime_str :: T.Text,
-              slotInstructor :: T.Text
-              } deriving Show
+    CourseSlot {
+                slotSection :: T.Text,
```
Comment (david-yz-liu) [inline hs/WebParsing/TimeTableParser.hs]: I'm not a fan of this indentation style - I think we should switch to a 4-space indent for containers whose first element starts on a new line. What do you think?

File: hs/WebParsing/PrerequisiteParsing.hs [hunk 2]
```diff
@@ -23,53 +23,53 @@ import qualified Data.Text as T
...
 toPrereq :: String -> T.Text
 toPrereq expr =
-  let (beforeExpr, courseExpr, afterExpr) = matchCourse expr
-  in case (beforeExpr, courseExpr, afterExpr) of
-     (_, "", "") -> ""
-     --guaranteed match
-     (_, course, "") -> T.pack course
-     (_, course, after) ->  T.concat [(T.pack course), " ", (toPrereq after)]
+    let (beforeExpr, courseExpr, afterExpr) = matchCourse expr
+    in case (beforeExpr, courseExpr, afterExpr) of
```
Comment (david-yz-liu) [inline hs/WebParsing/PrerequisiteParsing.hs]: The 3-space indentation isn't great. I think it makes more sense to start multi-line `in` blocks on a new line, indented 4 spaces.

File: hs/WebParsing/PrerequisiteParsing.hs [hunk 3]
```diff
@@ -23,53 +23,53 @@ import qualified Data.Text as T
...
-  let (beforeStr, delimStr, afterStr) = matchDelim str
-  in case (beforeStr, delimStr, afterStr) of
-    ("","","") -> [] --if (expr == "") then [] else [expr]
-    (before, "", "") -> [before++expr]
-    (before, ",", after) -> if (isntDelim after)
-                            then toPreExprs after (expr ++ before)
-                            else (expr++before):(toPreExprs after "")
-    (before, ";", after) -> (expr++before):(toPreExprs after "")
+    let (beforeStr, delimStr, afterStr) = matchDelim str
+    in case (beforeStr, delimStr, afterStr) of
```
Comment (david-yz-liu) [inline hs/WebParsing/PrerequisiteParsing.hs]: See comment below about `in`

File: hs/WebParsing/GraphConversion.hs [hunk 4]
```diff
@@ -170,32 +170,32 @@ labNodes' graph = return (labNodes graph)
...
-           "CSC446H1", "CSC448H1", "CSC454H1", "CSC456H1", "CSC458H1",
-           "CSC463H1", "CSC465H1", "CSC469H1", "CSC485H1", "CSC486H1",
-           "CSC488H1", "ECE489H1", "CSC490H1",
-           "MAT135H1", "MAT136H1", "MAT137Y1", "MAT157Y1", "MAT221H1",
-           "MAT223H1", "MAT240H1", "MAT235Y1", "MAT237Y1", "MAT257Y1",
-           "STA247H1", "STA255H1", "STA257H1"] >>=
-  labNodes' >>=
-  print
+    toGraph' ["CSC104H1", "CSC108H1", "CSC120H1", "CSC148H1", "CSC165H1",
+              "CSC200Y1", "CSC207H1", "CSC209H1", "CSC236H1", "CSC240H1",
```
Comment (david-yz-liu) [inline hs/WebParsing/GraphConversion.hs]: For long lists like this, I think it makes more sense to start the elements on a new line, indented 4 spaces (see final comment below)

File: hs/Utilities.hs [hunk 5]
```diff
@@ -1,14 +1,19 @@
 {-# LANGUAGE OverloadedStrings #-}
 
-module MakeElements where
+module Utilities (createTag, stylesheet, makeScript, makeForm, makeInput, makeA, jQuery, concatHtml, concatSVG, mdToHTML) where
```
Comment (david-yz-liu) [inline hs/Utilities.hs]: We should list exports by putting each on a new line, indented 4 spaces

File: hs/Utilities.hs [hunk 6]
```diff
@@ -1,14 +1,19 @@
...
 import qualified Text.Blaze.Svg11 as S
-
-insertSVG :: H.AttributeValue -> H.Html
-insertSVG src = H.object ! A.data_ src ! A.type_ "image/svg+xml" $ ""
+import Text.Markdown (markdown, def)
+import Data.Text.Lazy (Text)
+import Control.Monad.IO.Class  (liftIO)
+import qualified Data.Conduit.List as CL
+import qualified Data.Text.Internal as TI
+import Data.Conduit
```
Comment (david-yz-liu) [inline hs/Utilities.hs]: Please name the functions/types that are imported

File: hs/Svg/Parser.hs [hunk 7]
```diff
@@ -43,17 +43,17 @@ performParse :: String -- ^ The title of the graph.
...
-      let parsedGraph = parseGraph key graphFile
-          PersistInt64 keyVal = toPersistValue key
-      print "Graph Parsed"
-      insertElements parsedGraph
-      print "Graph Inserted"
-      let genGraphPath = graphPath ++ "gen/"
-      createDirectoryIfMissing True genGraphPath
-      buildSVG key M.empty (genGraphPath ++ show keyVal ++ ".svg") False
-      print "Success"
+    do graphFile <- readFile (graphPath ++ inputFilename)
```
Comment (david-yz-liu) [inline hs/Svg/Parser.hs]: We should start `do` blocks on a new line, indented 4 spaces.

File: hs/Scripts.hs [hunk 8]
```diff
@@ -5,59 +5,59 @@ module Scripts where
...
-                                           "static/js/graph/setup.js",
-                                           "static/js/post/update_post.js",
-                                           "static/js/graph/sidebar/sidebar_divs.js",
-                                           "static/js/graph/sidebar/sidebar_events.js",
-                                           "static/js/graph/sidebar/focus_descriptions.js",
-                                           "static/js/common/export/export.js"])
+graphScripts :: H.Html
+graphScripts = concatHtml (map makeScript["https://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js",
+                                          "https://ajax.googleapis.com/ajax/libs/jqueryui/1.10.4/jquery-ui.min.js",
+                                          "/static/js/common/modal.js",
```
Comment (david-yz-liu) [inline hs/Scripts.hs]: I changed the indentation here already in the `facebook-config` branch, which was recently merged in. You should probably do a merge from master to grab that.

File: hs/GridResponse.hs [hunk 9]
```diff
@@ -46,16 +47,4 @@ searchPanel =
...
-                                                   createTag H.span "course-info-code" "" ""
-                                                   createTag H.span "course-info-title" "" ""
-
-                                            H.h4 $ do
-                                                   createTag H.span "section-stats-section" "" ""
-                                                   createTag H.span "section-stats-instructor" "" ""
-
-                                            createTag H.p "section-stats-enrol" "" ""
-                            -}
+        createTag H.div "info" "row" ""
```
Comment (david-yz-liu) [inline hs/GridResponse.hs]: I'd prefer the file to end with a newline

File: hs/Database/CourseVideoSeed.hs [hunk 10]
```diff
@@ -21,5 +21,5 @@ seedVideo (code, videos) = do
 
 
 seedVideos :: IO ()
-seedVideos = runSqlite dbStr $ do
+seedVideos = runSqlite databasePath $ do
     mapM_ seedVideo courseVideos
```
Comment (david-yz-liu) [inline hs/Database/CourseVideoSeed.hs]: Can you please put a newline here?

File: hs/AboutResponse.hs [hunk 11]
```diff
@@ -6,17 +6,16 @@ import           Text.Blaze ((!))
...
+import Utilities (mdToHTML)
 
 aboutResponse :: Text -> ServerPart Response
 aboutResponse aboutContents =
    ok $ toResponse $
     masterTemplate "Courseography - About"
                 []
-                (do
-                    header "about"
+                (do header "about"
```
Comment (david-yz-liu) [inline hs/AboutResponse.hs]: I'd prefer keeping what we had before

File: hs/WebParsing/TimeTableParser.hs (hunk 1 above)
Comment (Ian-Stewart-Binks) [inline hs/WebParsing/TimeTableParser.hs]: I'm not certain that the elements of containers should start on newlines. This style here is inconsistent with the rest of the codebase (specifically Database/Tables.hs).

File: hs/WebParsing/PrerequisiteParsing.hs (hunk 2 above)
Comment (Ian-Stewart-Binks) [inline hs/WebParsing/PrerequisiteParsing.hs]: You're right. It is awkward. I'm going to make a change - what do you think about the indentation of case statement patterns?

File: hs/AboutResponse.hs (hunk 11 above)
Comment (Ian-Stewart-Binks) [inline hs/AboutResponse.hs]: OK, but that is inconsistent with every other `*Response.hs` file. Do you want me to change those as well? Should every `do` block change?

File: hs/Svg/Parser.hs (hunk 7 above)
Comment (Ian-Stewart-Binks) [inline hs/Svg/Parser.hs]: Should the `do` keyword be on a new line? Right now it either ends a line or starts one.

**Comment by Ian-Stewart-Binks** [comment]
It might make more sense to make a refactoring branch.

File: hs/Svg/Parser.hs (hunk 7 above)
Comment (david-yz-liu) [inline hs/Svg/Parser.hs]: Good question - let's generally start them on the previous line, *especially* after `=`

**Comment by david-yz-liu** [comment]
@Ian-Stewart-Binks I've only been making comments on things you changed (plus newlines). I'd rather you just address the comments in those sections, rather than making the style consistent everywhere. This can come later.

**Comment by Ian-Stewart-Binks** [comment]
Sure thing.

**Comment by david-yz-liu** [comment]
Thanks, @Ian-Stewart-Binks!

**Pull request merged**